components/org.wso2.carbon.identity.conditional.auth.functions.analytics/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/analytics/CallAnalyticsFunctionImpl.java

```diff
@@ -213,4 +215,31 @@ public void cancelled() {
         });
         JsGraphBuilder.addLongWaitProcess(asyncProcess, eventHandlers);
     }
+
+    private Map<String, Object> getPayloadDataMap(Map<String, Object> payloadData) {
+        Map<String, Object> payloadDataMap = new HashMap<>();
+        for (Map.Entry<String, Object> entry : payloadData.entrySet()) {
+            Object value = entry.getValue();
+            if (value instanceof Map) {
+                payloadDataMap.put(entry.getKey(), getPayloadDataMap((Map<String, Object>) value));
+            } else if (value instanceof List) {
+                payloadDataMap.put(entry.getKey(), processList((List<Object>) value));
+            } else {
+                payloadDataMap.put(entry.getKey(), value);
+            }
+        }
+        return payloadDataMap;
+    }
+
+    private List<Object> processList(List<Object> list) {
+        List<Object> resultList = new ArrayList<>();
+        for (Object item : list) {
+            if (item instanceof Map) {
```

Comment:
can `item` be a List also?

Reply:
This is possible. Handled with ecb4acd9b9e89de443c225c4f58df10206068ab1